ccp/src/cu/proving_thread/tests.rs

```diff
@@ -194,30 +201,35 @@ async fn dataset_creation_works_with_three_threads() {
 async fn cc_job_stopable() {
     let epoch = test::generate_epoch_params(4, 0xFF);
     let cu_id = test::generate_cu_id(4);
-    let (thread, actual_dataset, mut outlet) =
-        create_thread_init_dataset(2.into(), epoch, cu_id).await;
+    let mut ingredients = ThreadInitIngredients::create(2.into(), epoch, cu_id).await;
 
     let flags = RandomXFlags::recommended_full_mem();
-    thread
-        .run_cc_job(actual_dataset, flags, epoch, cu_id)
+    ingredients
+        .thread
+        .run_cc_job(ingredients.dataset, flags, epoch, cu_id)
         .await
         .unwrap();
 
     let handle = tokio::spawn(async move {
         let flags = RandomXFlags::recommended();
         let global_nonce_cu = ccp_utils::hash::compute_global_nonce_cu(&epoch.global_nonce, &cu_id);
 
-        while let Some(ToUtilityMessage::ProofFound(proof)) = outlet.recv().await {
-            let expected_result_hash = run_light_randomx(
-                global_nonce_cu.as_slice(),
-                proof.local_nonce.as_ref(),
-                flags,
-            );
-            assert!(expected_result_hash.meet_difficulty(&epoch.difficulty));
+        while let Some(message) = ingredients.to_utility.recv().await {
+            match message {
+                ToUtilityMessage::ProofFound(proof) => {
+                    let expected_result_hash = run_light_randomx(
+                        global_nonce_cu.as_slice(),
+                        proof.local_nonce.as_ref(),
+                        flags,
+                    );
+                    assert!(expected_result_hash.meet_difficulty(&epoch.difficulty));
+                }
+                _ => {}
+            }
         }
     });
 
```

Comment:
I think you should sleep here for some time, otherwise the `while let` loop above may be executed for 0 times, as you stop the working thread immediately.

Reply:
Or, as it is `cc_job_stopable`, the loop above may be simplified to not distract from the test's goal.

Reply:
Yep, it's about stopping a prover, but I can't remove this, b/c otherwise writing to channel internally fails.

Reply:
You may remove the check, just consume messages repeatedly.